### PR TITLE
 create logstreams using directories provided in the configuration

### LINKED ISSUE #2297: Use provided configuration when creating logstream in DistributedLogstreamService
When logstream service creation is moved to DefaultDistributedLogstream service, the existing configurations are partially ignored. Need to fix that.

### PR DESCRIPTION
**Pull request by deepthidevaki**
- Distributed log creates logstream services using the configuration provided. 

closes #2297 #2326

### CODE REVIEW

File: broker-core/src/main/java/io/zeebe/broker/clustering/base/raft/RaftPersistentConfigurationManagerService.java [hunk 1]
```diff
@@ -53,12 +54,13 @@ public void start(ServiceStartContext startContext) {
...
-    LogstreamConfig.putLogDirectory(
-        String.valueOf(configuration.getCluster().getNodeId()),
-        dataConfiguration.getDirectories().get(0));
+    service =
+        new StorageConfigurationManager(
+            configuration.getData().getDirectories(),
+            configuration.getData().getDefaultLogSegmentSize());
 
-    service = new RaftPersistentConfigurationManager(configuration.getData());
+    /* A hack so that DistributedLogstream primitive can create logs in this directory */
```
Comment (Zelldon) [inline broker-core/src/main/java/io/zeebe/broker/clustering/base/raft/RaftPersistentConfigurationManagerService.java]: What would be the solution to remove this hack?

File: logstreams/pom.xml [hunk 2]
```diff
@@ -123,6 +123,17 @@
       <artifactId>zeebe-transport</artifactId>
       <scope>test</scope>
     </dependency>
+
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
```
Comment (Zelldon) [inline logstreams/pom.xml]: Is this necessary?

File: logstreams/src/main/java/io/zeebe/distributedlog/StorageConfiguration.java [hunk 3]
```diff
@@ -28,44 +26,41 @@
...
+ *   <li>logstream segments
+ *   <li>log block index
+ *   <li>log block index snapshots
+ *   <li>stream processor state
  * </ul>
  */
-public class RaftPersistentConfiguration {
-  private final RaftConfigurationMetadata configuration = new RaftConfigurationMetadata();
+public class StorageConfiguration {
+  private final StorageConfigurationMetadata configuration = new StorageConfigurationMetadata();
```
Comment (Zelldon) [inline logstreams/src/main/java/io/zeebe/distributedlog/StorageConfiguration.java]: Why we still need this?

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
I have some questions on this maybe we can discuss them before we merge the changes :) 

I'm also not sure if we still need the persistence of the metadata, I currently assume not?

**Comment by deepthidevaki** [comment]
@Zelldon I updated the PR. We don't persist the configuration anymore.

File: broker-core/src/main/java/io/zeebe/broker/clustering/base/raft/RaftPersistentConfigurationManagerService.java [hunk 4]
```diff
@@ -59,7 +59,7 @@ public void start(ServiceStartContext startContext) {
             configuration.getData().getDirectories(),
             configuration.getData().getDefaultLogSegmentSize());
 
-    /* A hack so that DistributedLogstream primitive can create logs in this directory */
+    /* A temp solution so that DistributedLogstream primitive can create logs in this directory */
```
Comment (Zelldon) [inline broker-core/src/main/java/io/zeebe/broker/clustering/base/raft/RaftPersistentConfigurationManagerService.java]: :laughing:

**Review by ChrisKujawa — APPROVED** [review]
thanks :+1:

**Pull request merged**